github/repos_commits_test.go

```diff
@@ -380,7 +375,7 @@ func TestRepositoriesService_CompareCommits(t *testing.T) {
 	}
 
 	for _, sample := range testCases {
-		client, mux, _, teardown := setup()
+		client, mux, _ := setup(t)
```

Comment:
@alexandear - I'm noticing in this test that `setup(t)` is getting called in a loop which is definitely not what we want since the new cleanup only happens once after the entire test completes.

It looks like this test is going to need a version of `setup` that does expose the `teardown`.

Thoughts?